yt/yt/library/auth_server/oauth_token_authenticator.cpp

```diff
@@ -82,13 +83,29 @@ class TOAuthTokenAuthenticator
 
     TErrorOr<TAuthenticationResult> OnGetUserInfoImpl(const TOAuthUserInfoResult& userInfo)
     {
-        auto result = WaitFor(UserManager_->CreateUser(userInfo.Login));
-        if (!result.IsOK()) {
-            auto error = TError("Failed to create user")
-                << TErrorAttribute("name", userInfo.Login)
-                << std::move(result);
-            YT_LOG_WARNING(error);
-            return error;
+        if (Config_->CreateUserIfNotExists) {
```

Comment:
Can we maybe separate a helper base class that implements this method? Copy-paste is a bit annoying. But if it is much simpler this way, ok.

Reply:
It was copy-pasted already, I just adjusted the code

I think it is possible to extract it to some `oauth_authenticator_base.cpp`, but in a separate PR

Reply:
Even more, `oauth_cookie_authenticator.cpp` and `oauth_token_authenticator` are almost identical, with `cookie -> token` replacement
https://gist.github.com/Krock21/91a043f3697f0d5ee739d3b63db60d52

Reply:
Let's remove this copypaste in a separate PR